### PR TITLE
Featured item block

### PR DESCRIPTION
**Pull request by MalcolmVonMoJ**
New featured item block

### CODE REVIEW

File: mojblocks.php [hunk 1]
```diff
@@ -453,6 +453,25 @@ function mojblocks_register_blocks()
...
 		);
 	}
+
+    register_block_type(
+        'mojblocks/featured-item',
+        [
+            'editor_script' => 'mojblocks-editor-script',
+            'render_callback' => 'render_callback_featured_item_block',
+            'attributes' => [
+                'featuredDocumentHasDate' => [
```
Comment (RobLoweMOJ) [inline mojblocks.php:463]: suggest change to featuredItem for consistancy

File: mojblocks.php [hunk 2]
```diff
@@ -547,6 +569,25 @@ function mojblocks_extend_wp_api($data, $post, $context) {
         $data->data['featured_image_url'] = "";
     }
 
+    // Add summary to API
+    $summary = get_post_meta( get_the_ID(), 'post_summary', TRUE); // get the value from the meta field
```
Comment (RobLoweMOJ) [inline mojblocks.php:573]: Worth a discussion as we talking about adding something into hale that removes the need for this

File: src/custom-blocks/featured-item/edit.js [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+import { store as coreStore } from '@wordpress/core-data';
+
+const { Fragment } = wp.element;
+const d = new Date();
+const allowedMediaTypes = ['image'];
+const templateFeaturedDocumentBlock = [
+	[ 'core/heading', { placeholder: 'Add featured item section title' } ]
+];
+
+export default function FeaturedDocumentEdit({ attributes, setAttributes} ) {
```
Comment (RobLoweMOJ) [inline src/custom-blocks/featured-item/edit.js:25]: suggest FeaturedDocument to FeaturedItem

File: src/custom-blocks/featured-item/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    },
+    featuredDocumentID: {
+      type: "string",
+      default: "0"
+    },
+    featuredItemType: {
+      type: "string",
+      default: "post"
+    },
+    featuredImage: {
```
Comment (RobLoweMOJ) [inline src/custom-blocks/featured-item/index.js:32]: remove duplicate

Comment (MalcolmVonMoJ) [reply src/custom-blocks/featured-item/index.js:32]: Well spotted Rob!  Excellent reviewing!!! 😉  Done

File: mojblocks.php (hunk 2 above)
Comment (MalcolmVonMoJ) [reply mojblocks.php]: Done

File: src/custom-blocks/featured-item/edit.js [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+				allPostTypes: allPostTypeThingies
+			};
+		}
+	);
+
+	const {
+		allDocuments,
+	} = useSelect(
+		( select ) => {
+			if (featuredItemType != "" && featuredItemID.length > 0) {
```
Comment (RobLoweMOJ) [inline src/custom-blocks/featured-item/edit.js:60]: why is featuredItemID.length > 0 needed?   before you select wouldnt it be 0?

File: src/custom-blocks/featured-item/edit.js [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+
+	const {
+		allPostTypes,
+	} = useSelect(
+		( select ) => {
+
+			const { getPostTypes } = select(
+				coreStore
+			);
+			const allPostTypeThingies = getPostTypes(
```
Comment (RobLoweMOJ) [inline src/custom-blocks/featured-item/edit.js:47]: Thingies?

File: src/custom-blocks/featured-item/edit.js [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+				date: allDocuments[i].date,
+				imageID: allDocuments[i].featured_media,
+				image: allDocuments[i].featured_image_url,
+			}
+			docOptions.push({label: allDocuments[i].title.rendered, value: allDocuments[i].id});
+
+		}
+	}
+
+	/* Not used yet
```
Comment (RobLoweMOJ) [inline src/custom-blocks/featured-item/edit.js:123]: Remove if not used

File: src/custom-blocks/featured-item/edit.js [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+		setAttributes({
+			featuredCustomImage: null,
+		});
+	};
+
+	const setItemType = newItemType => {
+		setAttributes({ featuredItemType: newItemType });
+	};
+
+	const setDocument = newDocumentID => {
```
Comment (RobLoweMOJ) [inline src/custom-blocks/featured-item/edit.js:141]: setFeaturedItem

File: src/custom-blocks/featured-item/edit.js (hunk 5 above)
Comment (MalcolmVonMoJ) [reply src/custom-blocks/featured-item/edit.js]: I have test this without and checked the output - I don't think it is needed so I've removed it.

File: src/custom-blocks/featured-item/edit.js (hunk 6 above)
Comment (MalcolmVonMoJ) [reply src/custom-blocks/featured-item/edit.js]: Changed - isn't naming variables the hardest thing in the world!??!?

File: src/custom-blocks/featured-item/edit.js (hunk 8 above)
Comment (MalcolmVonMoJ) [reply src/custom-blocks/featured-item/edit.js]: Changed.  Also changed other references to "document" such as allDocuments

File: src/custom-blocks/featured-item/edit.js (hunk 7 above)
Comment (MalcolmVonMoJ) [reply src/custom-blocks/featured-item/edit.js:123]: Removed

**Review by RobLoweMOJ — APPROVED** [review]
LGTM

**Pull request merged**